lib/macho/load_commands.rb

```diff
@@ -205,14 +249,27 @@ def to_s
     # explicit operations on the raw Mach-O data.
     class LCStr
       # @param lc [MachO::LoadCommand] the load command
-      # @param lc_str [Fixnum] the offset to the beginning of the string
+      # @param lc_str [Fixnum, String] the offset to the beginning of the string,
+      #  or the string itself if not being initialized with a view.
+      # @todo devise a solution such that the `lc_str` parameter is not
+      #  interpreted differently depending on `lc.view`. The current behavior
+      #  is a hack to allow viewless load command creation.
       # @api private
       def initialize(lc, lc_str)
         view = lc.view
-        lc_str_abs = view.offset + lc_str
-        lc_end = view.offset + lc.cmdsize - 1
-        @string = view.raw_data.slice(lc_str_abs..lc_end).delete("\x00")
-        @string_offset = lc_str
+
+        if view
+          lc_str_abs = view.offset + lc_str
+          lc_end = view.offset + lc.cmdsize - 1
+          @string = view.raw_data.slice(lc_str_abs..lc_end).delete("\x00")
+          @string_offset = lc_str
+        else
+          # setting string_offset to the size of the command assumes that there
+          # is only one lc_str in the command. this is presently true for all
+          # load commands, but is not guaranteed in the future.
```

Comment:
I think this comment (no longer) belongs here. It's also not fully accurate either, as there's `prebound_dylib_command` that hosts two instances of an `lc_str`, thus multiple `lc_str`s are not just a future thing.